src/server/repository/assessment/section/getOne.ts

```diff
@@ -10,14 +10,15 @@ export const getOne = async (
   client: BaseProtocol = DB
 ): Promise<Section> => {
   const schemaName = Schemas.getName(props.assessment)
-  return client.one<Section>(
+
+  return client.oneOrNone<Section>(
     `
         select s.*
         from ${schemaName}.section s
         where ${'sectionName' in props ? `s.props->>'name' = $2` : `s.id = $2`}
           and props -> 'cycles' ? $1;
     `,
     [props.cycle.uuid, 'sectionName' in props ? props.sectionName : props.id],
-    SectionAdapter
+    (section) => section && SectionAdapter(section)
```

Comment:
does it call this if nothing is returned

Reply:
the method should be getOneOrOne and use client.oneOrNone in case that's what needed. but is it ?

Reply:
the return type of the method should be updated as well. but why do we want to allow this? 
shouldn't the caller know that it doesn't need to call this method ?

Reply:
the error was caused by the SectionAdapter.

I had two options: add an exception in case the sectionName is 'messageBoard' or allow that the api can return null in case a section has not been found.

Reply:
Section is not mandatory to create a message . Then you can avoid to call this method in such case?

Reply:
ok, fixed